src/webgpu/web-platform/reftests/gpu_ref_test.ts

```diff
@@ -0,0 +1,49 @@
+import { DevicePool } from '../../../common/framework/gpu/implementation.js';
+import { assert, unreachable } from '../../../common/framework/util/util.js';
+
+const devicePool = new DevicePool();
+
+export class GPURefTest {
+  private objects: { device: GPUDevice; queue: GPUQueue } | undefined = undefined;
+  initialized = false;
+
+  get device(): GPUDevice {
+    assert(this.objects !== undefined);
+    return this.objects.device;
+  }
+
+  get queue(): GPUQueue {
+    assert(this.objects !== undefined);
+    return this.objects.queue;
+  }
+
+  async init(): Promise<void> {
+
+    const device = await devicePool.acquire();
+    const queue = device.defaultQueue;
+    this.objects = { device, queue };
+
+    try {
+      await device.popErrorScope();
+      unreachable('There was an error scope on the stack at the beginning of the test');
+    } catch (ex) { }
+
+    device.pushErrorScope('out-of-memory');
+    device.pushErrorScope('validation');
```

Comment:
I don't think it's even very important to check for errors during this test. We'll have plenty of other tests where we can check errors (including tests which do about the same thing as this, but without the reftesting).